### PR TITLE
Ssr

### PR DESCRIPTION
**Pull request by manedurphy**
@munzmoses, hey brother, when you have a chance can you have a look at my server-side rendering code? I'm handling most of the rendering logic in the SSR class and instantiate it in the reviewsRender.js file.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    </div>
+    <div class="col-2">
+      <div id="app"></div>
+    </div>
+  </div>
+  <div id="footer"></div>
+</body>
+<script crossorigin src="https://unpkg.com/react@17/umd/react.production.min.js"></script>
+<script crossorigin src="https://unpkg.com/react-dom@17/umd/react-dom.production.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js"
```
Comment (munzmoses) [inline public/index.html:57]: nice dude! are you gonna use CDN's for all of your npm packages in the end?

File: server/classes/SSR.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import axios from 'axios';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
```
Comment (munzmoses) [inline server/classes/SSR.js:3]: gonna have to check this one out for sure.

File: server/classes/SSR.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        this.reviewsData = null;
+    }
+
+    getId() {
+        const splitUrl = this.baseUrl.split('/');
+        return splitUrl[splitUrl.length - 1];
+    }
+
+    async getReviewsData() {
+        const { data } = await axios.get(`${process.env.REVIEWS_DOMAIN}/api/reviews/all/${this.getId()}`);
```
Comment (munzmoses) [inline server/classes/SSR.js:19]: I notice you're using process.env here but not using the dotenv package. Is this linked up with your docker image?

File: server/index.js [hunk 4]
```diff
@@ -1,12 +1,9 @@
...
+import reviewsRender from './routes/reviewsRender';
 
 const app = express();
-
-app.use(...reviewsAllRoutes);
-app.use(...reviewCalculationRoutes);
-
 const port = process.env.PORT || 6002;
 
+app.use(...reviewsRender);
```
Comment (munzmoses) [inline server/index.js:7]: Interesting, so using reviewsRender is now taking care of all prior functionality? Is this essentially serving up your entire frontend as well?

File: server/routes/reviewsRender.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        fs.readFile(path.resolve(__dirname, '../../public/index.html'), 'utf8', async (err, data) => {
+            if (err) return res.status(500).json({ message: 'Internal server error' });
+
+            try {
+                const ssr = new SSR(req.baseUrl, data);
+                const html = await ssr.getHtml();
+
+                return res.send(html);
+            } catch (error) {
+                const noDataHtml = SSR.getHtmlWithNoData(data);
```
Comment (munzmoses) [inline server/routes/reviewsRender.js:17]: Great fallback method.

File: services/reviewService/components/Review.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react/prop-types */
+// this component will render individual reviews
+import React from 'react';
```
Comment (munzmoses) [inline services/reviewService/components/Review.jsx:3]: Did SSR require a refactor on the frontend as well? If so, was it a complete refactor or a minor refactor?

File: services/reviewService/components/Title.jsx [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import React from 'react';
+const Title = () => <h2 className="reviews-section-title">What people are saying</h2>;
```
Comment (munzmoses) [inline services/reviewService/components/Title.jsx:2]: It looks like this doesn't need its own file per se. This probably doesn't pertain to you, considering it's on the frontend and was most likely inherited.

**Review by munzmoses — APPROVED** [review]
It looks great man, didn't have much feedback because I'm not versed in the SSR game yet. Nice work fam!

File: public/index.html (hunk 1 above)
Comment (manedurphy) [reply public/index.html:57]: Yes, it is looking that way

File: server/classes/SSR.js (hunk 3 above)
Comment (manedurphy) [reply server/classes/SSR.js:19]: Yes, I set my ENVs in the docker containers, but if I run this locally, I use a plugin of my Oh-My-Zsh terminal that automatically puts variables in .env files into my environment. Super useful tool!

File: server/index.js (hunk 4 above)
Comment (manedurphy) [reply server/index.js:7]: Yes, this just inputs all the logic for rendering on the server. I think I will change this from "reviewsRender" to just render when I clone everyone's repos and add their frontends in.

File: server/routes/reviewsRender.js (hunk 5 above)
Comment (manedurphy) [reply server/routes/reviewsRender.js:17]: Thank you sir

File: services/reviewService/components/Review.jsx (hunk 6 above)
Comment (manedurphy) [reply services/reviewService/components/Review.jsx:3]: Not much. I just chose to clean up some extra stuff that was either causing errors or just made more sense it me. Using the renderToString method should work with any react component you throw in it.

File: services/reviewService/components/Title.jsx (hunk 7 above)
Comment (manedurphy) [reply services/reviewService/components/Title.jsx:2]: I agree. I will probably end up refactoring the frontend even more when I have down time.

**Pull request merged**